flocker/node/_docker.py

```diff
@@ -527,6 +528,50 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        """
+        Supply data about an image, by either inspecting it or returning
+        cached data if available.
+
+        :param unicode image: The ID of the image.
+
+        :param unicode container: The ID of the container for which this
+            image is being used.
+
+        :param unicode state: The state of the container.
+
+        :return: ``dict`` representing data about the image properties.
+        """
+        # This caching technique isn't thread-safe, but allowable
+        # in this case because a) the data is for our purposes
+        # immutable, so two writes of same key isn't a problem and
+        # b) we never remove entries.
+        if image in self._image_cache:
+            return self._image_cache[image]
```

Comment:
Cache hit and miss logging would make it possible to evaluate how effective this cache is on a real system.  Cache hit information could also help with debugging - for example, to reveal that the image data is stale because it's being served from a cache.